include/copc-lib/copc/file.hpp

```diff
@@ -13,23 +16,38 @@ const int COPC_OFFSET = 429;
 class CopcFile : public las::LasFile
 {
   public:
-    CopcFile(las::LasHeader header, las::CopcVlr copc, las::WktVlr wkt, las::EbVlr eb)
-        : LasFile(header, eb), copc(copc), wkt(wkt){};
-    CopcFile(las::LasHeader header, int span, std::string wkt, las::EbVlr eb) : LasFile(header, eb)
+    CopcFile(const las::LasHeader &header, const las::CopcInfoVlr &copc_info, const las::CopcExtentsVlr &copc_extents,
+             const las::WktVlr &wkt, const las::EbVlr &eb)
+        : LasFile(header, eb), copc_info_(copc_info), copc_extents_(copc_extents), wkt_(wkt){};
+    CopcFile(const las::LasHeader &header, int span, const std::string &wkt, const las::EbVlr &eb) : LasFile(header, eb)
     {
-        this->wkt.wkt = wkt;
-        this->copc.span = span;
+        this->wkt_.wkt = wkt;
+        this->copc_info_.span = span;
+        this->copc_extents_.items = std::vector<las::CopcExtent>(
```

Comment:
I wonder if we should make the Extent vector into a dictionary instead? So that we can set extent["Intensity"] etc instead of having to know the exact ordering.

Reply:
Or maybe we should make our own Statistics object where we can set each of those fields.

Reply:
Good idea, I like our own object better so it's easier to use in cpp as well.